### ISSUE TITLE
codeunit 1698 "Feature Bank Deposits": bug transferring Deposit Header to new Bank Deposit Header table

### ISSUE THREAD

**Issue by ekesteloot** [microsoft/ALAppExtensions#23792]
In the following code segment, we believe the code is missing a ChangeCompany:
```
    local procedure TransferDeposits(DepositsTableId: Integer; var Company: Record Company)
    var
        BankDepositHeader: Record "Bank Deposit Header";
        DepositHeaderRecRef: RecordRef;
        BankDepositHeaderRecRef: RecordRef;
        PreviousDepositNoFieldRef: FieldRef;
        PreviousDepositNo: Code[20];
    begin
        DepositHeaderRecRef.Open(DepositsTableId, false, Company.Name);
        if DepositHeaderRecRef.IsEmpty() then
            exit;
        DepositHeaderRecRef.FindSet();
        BankDepositHeaderRecRef.Open(Database::"Bank Deposit Header", false, Company.Name);
        repeat
            PreviousDepositNoFieldRef := DepositHeaderRecRef.Field(BankDepositHeader.FieldNo("No."));
            PreviousDepositNo := PreviousDepositNoFieldRef.Value();
            if not BankDepositHeader.Get(PreviousDepositNo) then begin
                BankDepositHeaderRecRef.Init();
                TransferFields(BankDepositHeaderRecRef, DepositHeaderRecRef);
                BankDepositHeaderRecRef.Insert();
            end;
        until DepositHeaderRecRef.Next() = 0;
        RemoveNAsDeposits(DepositsTableId, Company);
    end;
```

The BankDepositHeader record variable should switch companies before the Get operation:
```
    local procedure TransferDeposits(DepositsTableId: Integer; var Company: Record Company)
    var
        BankDepositHeader: Record "Bank Deposit Header";
        DepositHeaderRecRef: RecordRef;
        BankDepositHeaderRecRef: RecordRef;
        PreviousDepositNoFieldRef: FieldRef;
        PreviousDepositNo: Code[20];
    begin
        DepositHeaderRecRef.Open(DepositsTableId, false, Company.Name);
        if DepositHeaderRecRef.IsEmpty() then
            exit;
        DepositHeaderRecRef.FindSet();

        BankDepositHeader.ChangeCompany(Company.Name);  // New ChangeCompany code

        BankDepositHeaderRecRef.Open(Database::"Bank Deposit Header", false, Company.Name);
        repeat
            PreviousDepositNoFieldRef := DepositHeaderRecRef.Field(BankDepositHeader.FieldNo("No."));
            PreviousDepositNo := PreviousDepositNoFieldRef.Value();
            if not BankDepositHeader.Get(PreviousDepositNo) then begin
                BankDepositHeaderRecRef.Init();
                TransferFields(BankDepositHeaderRecRef, DepositHeaderRecRef);
                BankDepositHeaderRecRef.Insert();
            end;
        until DepositHeaderRecRef.Next() = 0;
        RemoveNAsDeposits(DepositsTableId, Company);
    end;
```

**Comment by ekesteloot** [comment]
Please Note:  we filed this issue here, because we could not find the appropriate site in Partner Center to report the issue.  If you'd like to redirect us, we'd be happy to report the issue there.

**Comment by AndersLarsenMicrosoft** [comment]
We determined this was not an Extensibility Bug, but a HF request.

The impact of the bug is really low:
it's only for v22s (v23 and master don't have that code).
only for NA.
only affects if they have a Deposit with the same "No." in another company.
they can easily unblock themselves by removing/posting in-progress Deposits before enabling the feature.

We decided not to fix it a an extensibility request, if they really need it they should go through the HF request process.
